lxd/backup/backup_config_utils.go

```diff
@@ -38,11 +41,34 @@ func ConfigToInstanceDBArgs(c *config.Config, projectName string) *db.InstanceAr
 		Ephemeral:    c.Container.Ephemeral,
 		LastUsedDate: c.Container.LastUsedAt,
 		Name:         c.Container.Name,
-		Profiles:     c.Container.Profiles,
 		Stateful:     c.Container.Stateful,
 	}
 
-	return inst
+	if applyProfiles {
+		err := state.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
+			inst.Profiles = make([]api.Profile, 0, len(c.Container.Profiles))
+			for _, profileName := range c.Container.Profiles {
+				profile, err := cluster.GetProfileIfEnabled(ctx, tx.Tx(), projectName, profileName)
```

Comment:
We could avoid querying the project features for every profile (via `ProjectHasProfiles` inside `GetProfileIfEnabled`) if we did it ourselves once here, and then called `GetProfile` with the right project instead.